packages/react/src/Trans.tsx

```diff
@@ -35,8 +36,23 @@ export type TransProps = {
 } & TransRenderCallbackOrComponent
 
 export function Trans(props: TransProps): React.ReactElement<any, any> | null {
-  const { i18n, defaultComponent } = useLingui()
-  const { render, component, id, message, formats } = props
+  const lingui = useLingui()
+  return React.createElement(TransNoContext, { ...props, lingui })
+}
+
+export function TransNoContext(
```

Comment:
https://github.com/lingui/js-lingui/issues/1698#issuecomment-1655931468

Reply:
> Do we need a new component? Is this going to be part of the public api?

Yes. Yes.

The new component might not be used directly, but as part of the nextjs integration. But because we could not predict all cases upfront it's better to make it as part of a public api.  Having this, you can start directly using it in RSC and next app router without waiting for other changes to the lib or special integrations.  

> Can we distinguish RSC / client component at the useLingui level so that the current Trans works for both?

You don't have information is it executed in RSC or just SSR (client components) in runtime. Or i didn't find that. 

The only way of detecting RSC i've found in the compile / webpack level. Which is not ideal. So some of the users may want to avoid compile time transformations and use NoContext version directly. For others we could create a nextjs plugin which would do a heavy lifting under the hood (see my explanation comment)

I'm opened for discussion / brainstorming how to support RSC better.